### PR TITLE
Dbuhay/controllers tests

### PR DESCRIPTION
**Pull request by Diarsid**

### CODE REVIEW

**Review by Sameperson — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -88,6 +88,20 @@
...
             <version>1.7.21</version>
         </dependency>
+        <!-- REST assured -->
+        <dependency>
+            <groupId>io.rest-assured</groupId>
+            <artifactId>rest-assured</artifactId>
+            <version>3.0.2</version>
+            <scope>test</scope>
+        </dependency>
+        <!-- JSONSimple to create test JSON data easily -->
```
Comment (loveforjava) [inline pom.xml:98]: plz, change comment

File: src/main/java/com/opinta/controller/PostOfficeController.java [hunk 2]
```diff
@@ -46,8 +46,12 @@ public PostOfficeController(PostOfficeService postOfficeService) {
 
 	@PostMapping
     @ResponseStatus(HttpStatus.OK)
-	public void createPostOffice(@RequestBody PostOfficeDto postOfficeDto) {
-		postOfficeService.save(postOfficeDto);
+	public ResponseEntity<?> createPostOffice(@RequestBody PostOfficeDto postOfficeDto) {
+		postOfficeDto = postOfficeService.save(postOfficeDto);
+		if ( postOfficeDto == null && postOfficeDto.getId() <= 0 ) {
```
Comment (loveforjava) [inline src/main/java/com/opinta/controller/PostOfficeController.java:51]: Plz, delete redundant indents

Comment (loveforjava) [inline src/main/java/com/opinta/controller/PostOfficeController.java:51]: Here we can get NullPointerException

File: src/main/java/com/opinta/controller/AddressController.java [hunk 3]
```diff
@@ -45,8 +46,12 @@ public AddressController(AddressService addressService) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.OK)
-    public void createAddress(@RequestBody AddressDto addressDto) {
-        addressService.save(addressDto);
+    public ResponseEntity<?> createAddress(@RequestBody AddressDto addressDto) {
+        addressDto = addressService.save(addressDto);
+        if (addressDto == null) {
+            return new ResponseEntity<>("Failed to create new Address using given data.", HttpStatus.BAD_REQUEST);
```
Comment (loveforjava) [inline src/main/java/com/opinta/controller/AddressController.java:52]: Plz, check other controllers for response status and make changes through your class appropriately.

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
+package integration;
+
+import org.json.simple.JSONObject;
+import org.junit.Test;
+
+import static javax.servlet.http.HttpServletResponse.SC_NOT_FOUND;
+import static javax.servlet.http.HttpServletResponse.SC_OK;
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:7]: Plz, change to spring HTTP status

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+                .then()
+                .body("id", equalTo(1));
+    }
+    
+    @Test
+    public void getAddress_notFound() throws Exception {
+        expect()
+                .statusCode(404)
+                .when()
+                .get("/addresses/{id}", 764563);
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:37]: Plz, get rid of hardcoding

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                .when()
+                .get("/addresses/{id}", 1)
+                .then()
+                .body("id", equalTo(1));
+    }
+    
+    @Test
+    public void getAddress_notFound() throws Exception {
+        expect()
+                .statusCode(404)
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:35]: Plz make it constant

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+    
+        expect()
+                .statusCode(SC_OK)
+                .when()
+                .get("addresses/{id}", newAddrId);
+    }
+    
+    @Test
+    public void updateAddress() throws Exception {
+
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:72]: Plz, delete redundant empty line

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        updatedAddr.put("appartmentNumber", "32");
+        updatedAddr.put("description", "none");
+    
+        String newDistrict = given()
+                .contentType("application/json;charset=UTF-8")
+                .body(updatedAddr.toJSONString())
+                .expect()
+                .statusCode(SC_OK)
+                .when()
+                .put("/addresses/{id}", updatedId)
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:91]: Why do you think you have the appropriate address in your db?

File: src/test/java/integration/AddressControllerIntegrationTest.java [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                .when()
+                .delete("/addresses/{id}", newAddrId);
+    
+        expect()
+                .statusCode(SC_NOT_FOUND)
+                .when()
+                .get("/addresses/{id}", newAddrId);
+    }
+    
+}
```
Comment (loveforjava) [inline src/test/java/integration/AddressControllerIntegrationTest.java:140]: Plz, add empty line

File: src/test/java/integration/PostOfficeControllerIntegrationTest.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+                .statusCode(SC_OK)
+                .when()
+                .get("/post-offices/{id}", updatedId)
+                .then()
+                .body("name", equalTo(newName));
+    }
+    
+    @Test
+    public void deletePostOffice() throws Exception {
+
```
Comment (loveforjava) [inline src/test/java/integration/PostOfficeControllerIntegrationTest.java:96]: plz, delete empty line

File: src/test/java/integration/PostOfficeControllerIntegrationTest.java [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+                .when()
+                .get("/post-offices/{id}", updatedId)
+                .then()
+                .body("name", equalTo(newName));
+    }
+    
+    @Test
+    public void deletePostOffice() throws Exception {
+        
+        // create PostcodePool
```
Comment (loveforjava) [inline src/test/java/integration/PostOfficeControllerIntegrationTest.java:97]: Plz, or do comments like this everywhere or nowhere

**Pull request closed without merging**